spring-cloud-aws-autoconfigure/src/test/java/io/awspring/cloud/autoconfigure/config/AwsConfigurerClientConfiguration.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2013-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.awspring.cloud.autoconfigure.config;
+
+import io.awspring.cloud.autoconfigure.config.parameterstore.AwsClientConfigurerParameterStore;
+import io.awspring.cloud.autoconfigure.config.secretsmanager.AwsClientConfigurerSecretsManager;
+import java.time.Duration;
+import org.springframework.boot.BootstrapRegistry;
+import org.springframework.boot.BootstrapRegistryInitializer;
+import org.springframework.lang.Nullable;
+import software.amazon.awssdk.core.client.config.ClientOverrideConfiguration;
+import software.amazon.awssdk.http.SdkHttpClient;
+import software.amazon.awssdk.http.apache.ApacheHttpClient;
+import software.amazon.awssdk.services.secretsmanager.SecretsManagerClientBuilder;
+import software.amazon.awssdk.services.ssm.SsmClientBuilder;
+
+public class AwsConfigurerClientConfiguration implements BootstrapRegistryInitializer {
+
+	@Override
+	public void initialize(BootstrapRegistry registry) {
+		registry.register(AwsClientConfigurerSecretsManager.class,
```

Comment:
@maciejwalkowiak We can't use `AwsClientConfigurer.class` here since it would be overwritten by Parameter store integration. This is why I created `AwsClientConfigurerParameter` and `AwsClientConfigurerSecretsManager`. Which I can reference more easily.